### PR TITLE
Add Data Processing Plan pages

### PR DESCRIPTION
**Pull request by bjss-colin**

### CODE REVIEW

File: src/NHSD.GPIT.BuyingCatalogue.Services/Orders/Contracts/DataProcessingPlanService.cs [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+        await dbContext.SaveChangesAsync();
+
+        return dataProcessingPlan;
+    }
+
+    public async Task<DataProcessingPlan> GetDefaultDataProcessingPlan() =>
+        await dbContext
+            .DataProcessingPlans
+            .Include(d => d.Steps)
```
Comment (magnusjohn-bjss) [inline src/NHSD.GPIT.BuyingCatalogue.Services/Orders/Contracts/DataProcessingPlanService.cs:34]: Shouldn't this include the category as well?

File: src/NHSD.GPIT.BuyingCatalogue.Services/TaskList/TaskListService.cs [hunk 2]
```diff
@@ -214,6 +216,7 @@ private async Task<OrderTaskListCompletedSections> GetOrderSectionFlags(int orde
                 FundingInProgress = order.OrderItems.Any(oi => oi.OrderItemFunding != null),
                 FundingCompleted = order.OrderItems.All(oi => oi.OrderItemFunding != null),
                 HasImplementationPlan = contract?.ImplementationPlanId != null,
+                DataProcessingPlanCompleted = await dbContext.Contracts.AnyAsync(p => p.OrderId == orderId && p.DataProcessingPlan != null),
```
Comment (magnusjohn-bjss) [inline src/NHSD.GPIT.BuyingCatalogue.Services/TaskList/TaskListService.cs]: There's a local copy of the contract here, no need to go back to the db

Comment (ghost) [reply src/NHSD.GPIT.BuyingCatalogue.Services/TaskList/TaskListService.cs]: Good spot, missed this as part of the rebase.

File: src/NHSD.GPIT.BuyingCatalogue.Services/Orders/Contracts/DataProcessingPlanService.cs [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+            .FirstOrDefaultAsync(d => d.IsDefault);
```
Comment (magnusjohn-bjss) [inline src/NHSD.GPIT.BuyingCatalogue.Services/Orders/Contracts/DataProcessingPlanService.cs:36]: Should this include the category as well?

**Comment by bjss-colin** [comment]
Converted to a draft as there is a discrepancy between this and the next story.

**Review by magnusjohn-bjss — APPROVED** [review]

File: src/NHSD.GPIT.BuyingCatalogue.Services/Orders/Contracts/DataProcessingPlanService.cs (hunk 3 above)
Comment (ghost) [reply src/NHSD.GPIT.BuyingCatalogue.Services/Orders/Contracts/DataProcessingPlanService.cs:36]: Added in place. Will see first what the plan is with categories and steps and whether the plan was to add these as part of this or another story.

**Pull request merged**